java-security/src/main/java/com/sap/cloud/security/token/validation/validators/XsuaaJwtSignatureValidator.java

```diff
@@ -54,12 +56,10 @@ private PublicKey fetchPublicKey(Token token, JwtSignatureAlgorithm algorithm) t
             throw new IllegalArgumentException("Token does not contain the mandatory " + KID_PARAMETER_NAME + " header.");
         }
 
-        String jwksUri = configuration.isLegacyMode() ? configuration.getUrl() + "/token_keys" : token.getHeaderParameterAsString(JKU_PARAMETER_NAME);
-        if (jwksUri == null) {
-            throw new IllegalArgumentException("Token does not contain the mandatory " + JKU_PARAMETER_NAME + " header.");
-        }
-
+        String jwksUri = configuration.isLegacyMode() ? configuration.getUrl() + "/token_keys" : configuration.getProperty(UAA_DOMAIN) + "/token_keys";
```

Comment:
Do we need a fallback to "url" property from configuration if UAA_DOMAIN is not available? 
We are supposed to add the fallback for XSA configuration support. Or would XSA configurations have isLegacyMode() return true anyways?